zebra-utils/src/bin/zebra-checkpoints/main.rs

```diff
@@ -95,12 +95,18 @@ fn main() -> Result<()> {
         .map(BlockHeight)
         .expect("zcashd has some mature blocks: wait for zcashd to sync more blocks");
 
+    let mut starting_height = BlockHeight(args::Args::from_args().last_checkpoint);
+    if starting_height.0 > 0 {
+        starting_height.0 += 1;
+    }
+    assert!(starting_height < height_limit);
```

Comment:
This check is a good idea, thanks for thinking of it.

Questions:
* If the user's zcashd hasn't synced up to the last checkpoint, should the tool panic? (yes?)
* If we've recently updated the checkpoint list, should the tool exit successfully, without printing any checkpoints? (panic or success are both ok?)

I think it's ok to panic when we're sure there are no new checkpoints. But let's provide a useful message:
```suggestion
    assert!(starting_height < height_limit, "No mature blocks after the last checkpoint: wait for zcashd to sync more blocks");
```

Note: This check isn't enough to make sure we generate checkpoints. For that, we need to make sure that zcashd has at least `MAX_CHECKPOINT_HEIGHT_GAP` extra mature blocks. (But it could generate checkpoints earlier, if the blocks are bigger than `MAX_CHECKPOINT_BYTE_COUNT`.)

But since this is an internal tool, I think it's good to panic on the unexpected case, where zcashd hasn't fully synced. And I think it's ok to panic if we just did an update.

I also think it's ok to sometimes generate no extra checkpoints, and exit successfully.